src/hooks/useIsMouseEnter.ts

```diff
@@ -1,12 +1,13 @@
 import { useCallback, useState } from "react";
 
 import useEventListener from "./useEventListener";
+import { RefObject } from "react";
 
-export default function useMergeRefs({ ref }) {
-  const [isHovered, setIsHover] = useState(false);
+export default function useMergeRefs({ ref }: { ref: RefObject<HTMLElement> }) {
+  const [isHovered, setIsHover] = useState<boolean>(false);
```

Comment:
I think TS can defer the type 
```suggestion
  const [isHovered, setIsHover] = useState(false);
```